src/module/item/helpers.ts

```diff
@@ -74,4 +82,91 @@ function markdownToHTML(markdown: string): string {
         .innerHTML.trim();
 }
 
-export { itemIsOfType, markdownToHTML, performLatePreparation, reduceItemName };
+const valueRegex = /^([\w-]+)-(\d{1,3})$/;
+
+function evaluateTraitData(traits?: string[]): ValuedTraits {
+    if (!traits) return {};
+
+    const data: ValuedTraits = {};
+    for (const trait of traits) {
+        const match = valueRegex.exec(trait);
+        if (match) {
+            const key = sluggify(match[1]);
+            data[key] = Math.max(data[key] ?? 0, Number(match[2]));
+        }
+    }
+    return data;
+}
+
+/**
+ * Wraps over item data to extend for the purposes of system data.
+ * @todo once all items use data models, replace with data model implementation
+ */
+function wrapItemTraits<T extends ItemTraits | ItemTraitsNoRarity | RarityTraitAndOtherTags | OtherTagsOnly>(
+    item: ItemPF2e,
+    traits: T,
+): T {
+    if (traits.value) {
+        traits.value = traits.value.filter((t) => t in item.constructor.validTraits);
+    }
```

Comment:
```suggestion
    traits.value &&= traits.value.filter((t) => t in item.constructor.validTraits);
```

Reply:
That's not what the error says

Reply:
Those should really be equivalent, but that's on typescript. And maybe we should iterate through them and splice out bad ones? It's not often that any are going to need to be removed.